### PR TITLE
DBS: Normalformen

### PR DESCRIPTION
**Pull request by Steve2955**

### CODE REVIEW

**Comment by Steve2955** [comment]
## Command results
<details>
<summary>Details: </summary>

<details>
<summary><em>Run doctoc</em></summary>

```Shell
changed:
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/ABWL-LEDER/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/DBS-DORENDORF/README.md
unchanged:
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/ASM-GUENTHER/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/PRO-KASCHE/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/SWE-KASCHE/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/SWS-KUSCHE/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/EMB-GUENTHER/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/OOP-CASSELT/README.md
```



</details>

</details>

## Changed files
<details>
<summary>Changed 2 files: </summary>

- ABWL-LEDER/README.md
- DBS-DORENDORF/README.md

</details>

<hr>

[:octocat: Repo](https://github.com/technote-space/toc-generator) | [:memo: Issues](https://github.com/technote-space/toc-generator/issues) | [:department_store: Marketplace](https://github.com/marketplace/actions/toc-generator)

File: DBS-DORENDORF/README.md [hunk 1]
```diff
@@ -141,6 +146,41 @@ Die Beziehungen können unterschiedliche Kardinalitäten besitzen:
...
+- Listen, Arrays müssen in eigene Relationen ausgelagert werden (wenn DBMS keine entsprechende Erweiterung bietet)
+
+### 2. Normalform
+
+- **Funktionale Abhängigkeit:** Attribut aus einer Menge $A$, kann stets einem Attribut der Menge $B$ zugeordnet werden; Umkehrung muss nicht gelten
+  - z.B. Matrikelnummer $\rightarrow$ Name des Studierenden
+- **Vollfunktionale Abhängigkeit:** Schlüssel als Kombination aus mehreren Attributen
+- Vorliegen der ersten Normalform + alle nicht zu den Schlüsselattributen der Relation gehörenden Attribute sind nur vom gesamten Primärschlüssel der Relation abhängig
+  - Relationen, die keinen zusamengesetzten Primärschlüssel besitzen entsprechen immer der zweiten Normalform
+- bei Verletzung können Einfüge-, Lösch- und Änderungsanomalien auftretten
```
Comment (RvNovae) [inline DBS-DORENDORF/README.md:168]:
```suggestion
- bei Verletzung können Einfüge-, Lösch- und Änderungsanomalien auftreten
```

**Review by RvNovae — APPROVED** [review]

File: DBS-DORENDORF/README.md [hunk 2]
```diff
@@ -141,6 +146,41 @@ Die Beziehungen können unterschiedliche Kardinalitäten besitzen:
...
+- Vorliegen der zweiten Normalform + kein Nicht-Schlüsselattribut ist von einem anderen Nicht-Schlüsselattribut funktional abhängig
+
+> Durch Normalisierung entsteht eine sehr große Anzahl an Tabellen $\rightarrow$ hoher Verarbeitungsaufwand durch Join-Operationen.
+> Beim Entwurf werden daher in der Regel einzelne Normalformen ignoriert $\rightarrow$ gezielte Denormalisierung (möglich durch Vermeidung von Anomalien, wenn nur lesend zugegriffen wird)
+
+### Allgemeine Design-Regeln für den Entwurf relationaler Datenbankschemata
+
+- jede Tabelle (Relation) sollte intuitiv einem Objekt oder Beziehungstyp entsprechen (aussagekräftige Namen $\rightarrow$ eindeutige Semantik der Relation)
+- Entwurf der Basistabellen unter Einhaltung der dritten Normalform $\rightarrrow$ keine Anomalien
+- Attribute die NULL-Werte zulassen sollten minimiert werden
```
Comment (nukerxy) [inline DBS-DORENDORF/README.md:181]:
```suggestion
- Attribute, die NULL-Werte zulassen sollten minimiert werden
```

**Review by nukerxy — CHANGES_REQUESTED** [review]
Danke fürs Aufschreiben

**Review by nukerxy — APPROVED** [review]

**Pull request merged**